core/client/src/main/java/alluxio/client/block/BlockMasterClient.java

```diff
@@ -18,17 +18,36 @@
 import alluxio.wire.WorkerInfo;
 
 import java.io.IOException;
+import java.net.InetSocketAddress;
 import java.util.List;
 
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * A client to use for interacting with a Block Master.
+ * A client to use for interacting with a block master.
  */
 @ThreadSafe
 public interface BlockMasterClient extends Client {
 
   /**
+   * Factory for {@link BlockMasterClient}.
+   */
+  class Factory {
+
+    private Factory() {} // prevent instantiation
+
+    /**
+     * Factory method for {@link BlockMasterClient}.
+     *
+     * @param masterAddress the master address
+     * @return a new {@link BlockMasterClient} instance
+     */
+    public static BlockMasterClient create(InetSocketAddress masterAddress) {
```

Comment:
Would it be possible to use the factory throughput and make the constructors private? Its unclear when one should be used over the other